bpo-39939: Add str.removeprefix and str.removesuffix (#18939)
python/cpython: Doc/whatsnew/3.9.rst

:class:`dict` class.  See :pep:`584` for a full description.
(Contributed by Brandt Bucher in :issue:`36144`.)

removeprefix and removesuffix methods for strings

> -------------------------------------------------

Reviewer: please add an empty line after the title.

> ``str.cutprefix(prefix)`` and ``str.cutsuffix(suffix)`` have been added to

Author: ```suggestion
``str.removeprefix(prefix)`` and ``str.removesuffix(suffix)`` have been added to
```

> easily remove an unneeded prefix or a suffix from a string. Corrsponding

Reviewer 2: ```suggestion
easily remove an unneeded prefix or a suffix from a string. Corresponding
```

> ``bytes`` and ``bytearray`` methods have also been added.

Reviewer: collections.UserString as well, no?

(Contributed by Dennis Sweeney in :issue:`18939`.)

> 

Reviewer: You should mention the PEP 616 somewhere.

For example: `See :pep:`616` for a full description.`

:class:`dict` class.  See :pep:`584` for a full description.
(Contributed by Brandt Bucher in :issue:`36144`.)

> removeprefix and removesuffix methods for strings

Reviewer: In What's New in Python 3.8, PEP numbers are not well advertised. I'm not sure if it's intentational or not. Maybe add it in the title?

```suggestion
PEP 616: New removeprefix() and removesuffix() string methods
```

-------------------------------------------------

> ``str.removeprefix(prefix)`` and ``str.removesuffix(suffix)`` have been added

Reviewer: I'm not sure, but I think that you can add links to the doc with this syntax:

```suggestion
:meth:`str.removeprefix(prefix) <str.removeprefix>` and :meth:`str.removesuffix(suffix) <str.removesuffix>` have been added
```

to easily remove an unneeded prefix or a suffix from a string. Corresponding
``bytes``, ``bytearray``, and ``collections.UserString`` methods have also been

> added. See :pep:`616` for a full description (Contributed by Dennis Sweeney in

Reviewer: ```suggestion
added. See :pep:`616` for a full description. (Contributed by Dennis Sweeney in
```

:class:`dict` class.  See :pep:`584` for a full description.
(Contributed by Brandt Bucher in :issue:`36144`.)

PEP 616: New removeprefix() and removesuffix() string methods
-------------------------------------------------------------

:meth:`str.removeprefix(prefix)<str.removeprefix>` and
:meth:`str.removesuffix(suffix)<str.removesuffix>` have been added
to easily remove an unneeded prefix or a suffix from a string. Corresponding
``bytes``, ``bytearray``, and ``collections.UserString`` methods have also been
added. See :pep:`616` for a full description. (Contributed by Dennis Sweeney in

> :issue:`18939`.)

Reviewer 3: This is not [18939](https://bugs.python.org/issue18939), which is about "Venv docs regarding original python install". Shouldn't this be [39939](https://bugs.python.org/issue39939)?

Reviewer: Right, it's a typo error: @sweeneyde: can you please propose a PR to fix the typo?

Reviewer: Or @elazarg: Do you want to propose a PR to fix the typo?

Reviewer 3: Sure. I thought it might be overkill :)

Author: Oops -- It looks like that's the GitHub PR number rather than the bpo number. I can't make a PR tonight so feel free to change it. If not, I can fix it tomorrow.

Reviewer 3: Done.
